core/iwasm/common/wasm_memory.c

```diff
@@ -147,23 +147,70 @@ wasm_runtime_free_internal(void *ptr)
 void *
 wasm_runtime_malloc(unsigned int size)
 {
+#if 0
     if (size == 0) {
         LOG_WARNING("warning: wasm_runtime_malloc with size zero\n");
         /* At lease alloc 1 byte to avoid malloc failed */
         size = 1;
     }
 
     return wasm_runtime_malloc_internal(size);
+#endif
+    return wasm_gc_malloc(size);
 }
 
 void *
 wasm_runtime_realloc(void *ptr, unsigned int size)
 {
+#if 0
     return wasm_runtime_realloc_internal(ptr, size);
+#endif
+    return NULL;
 }
 
 void
 wasm_runtime_free(void *ptr)
+{
+#if 0
+  wasm_runtime_free_internal(ptr);
+#endif
+  wasm_gc_free(ptr);
+}
+
+#if WASM_ENABLE_GC != 0
+static inline void *
+wasm_gc_malloc_internal(unsigned int size)
+{
+    return mem_allocator_malloc_with_gc(pool_allocator, size);
+}
+ 
+void *
+wasm_gc_malloc(unsigned int size)
+{
+    if (size == 0) {
+        LOG_WARNING("warning: wasm_runtime_malloc with size zero\n");
+        /* At lease alloc 1 byte to avoid malloc failed */
+        size = 1;
+    }
+
+    return wasm_gc_malloc_internal(size);
+}
+
+#if WASM_GC_MANUALLY != 0
+static inline void
+wasm_gc_free_internal(void *ptr)
+{
+    if (!ptr) {
+        LOG_WARNING("warning: wasm_gc_free with NULL pointer\n");
+        return;
+    }
+    mem_allocator_free_with_gc(pool_allocator, ptr);
+}
+
+void
+wasm_gc_free(void *ptr)
 {
     wasm_runtime_free_internal(ptr);
 }
+#endif // WASM_GC_MANUALLY
```

Comment:
Could you use `/* */` to comment the code, here it is C source and we normally use `/* */`